core/src/SYCL/Kokkos_SYCL_Parallel_Reduce.hpp

```diff
@@ -61,26 +61,25 @@ namespace Impl {
 namespace SYCLReduction {
 template <class ValueJoin, class ValueOps, typename WorkTag, typename ValueType,
           typename ReducerType, typename FunctorType, int dim>
-void workgroup_reduction(sycl::nd_item<dim>& item,
-                         sycl::local_ptr<ValueType> local_mem,
-                         ValueType* results_ptr,
-                         ValueType* device_accessible_result_ptr,
-                         const unsigned int value_count,
-                         const ReducerType& selected_reducer,
-                         const FunctorType& functor, bool final) {
+std::enable_if_t<FunctorValueTraits<ReducerType, WorkTag>::StaticValueSize == 0>
```

Comment:
If you introduce something like
```C++
template <class FunctorValueTraits>
inline constexpr bool use_shuffle_based_algorithm = FunctorValueTraits::StaticValueSize > 0;
```
it would improve readability.

Reply:
Ping